rust/nasl-interpreter/src/built_in_functions/misc.rs

```diff
@@ -62,6 +64,66 @@ pub fn unixtime(_: &str, _: &dyn Sink, _: &Register) -> Result<NaslValue, Functi
     }
 }
 
+/// Compress given data with gzip, when headformat is set to 'gzip' it uses gzipheader.
+pub fn gzip(_: &str, _: &dyn Sink, register: &Register) -> Result<NaslValue, FunctionError> {
+    let data = match register.named("data") {
+        Some(ContextType::Value(NaslValue::Null)) => return Ok(NaslValue::Null),
+        Some(ContextType::Value(x)) => Vec::<u8>::from(x),
+        _ => return Err(FunctionError::new("gzip", ("data").into())),
+    };
+    let headformat = match register.named("headformat") {
+        Some(ContextType::Value(NaslValue::String(x))) => x,
+        _ => "noheaderformat",
+    };
+
+    match headformat.to_string().eq_ignore_ascii_case("gzip") {
+        true => {
+            let mut e = GzEncoder::new(Vec::new(), Compression::default());
+            match e.write_all(&data) {
```

Comment:
mhm I don't like the repetition everything after new call is identical. Maybe there is a better way.

Reply:
I looked into it and thought, that `write_all` is a function of the `Write` trait. Maybe there is a way of utilizing this?